### ISSUE TITLE
 Error generating cuda files?

### ISSUE THREAD

**Issue by mattmakesthings** [leggedrobotics/darknet_ros#26]
Hi I'm running into an issue when building using  

catkin build darknet_ros -DCMAKE_BUILD_TYPE=Release

darknet_ros_msgs builds just fine but when it gets to darknet_ros I get this error
![image](https://user-images.githubusercontent.com/25335149/29225432-1d94db0a-7e83-11e7-9602-166fe08b1f1f.png)

any help would be appreciated :)

**Comment by mbjelonic** [comment]
Hi,

Can you please tell me the GPU that you are using? It might be a problem with the compute capability (version). This means that you need to check the compute capability (version) of your GPU. You can find a list of supported GPUs in CUDA here: https://en.wikipedia.org/wiki/CUDA#Supported_GPUs. Simply find the compute capability of your GPU and replace in darknet_ros/CMakeLists.txt the version number 61 with your version number:

-gencode arch=compute_61,code=sm_61

Best,
Marko

**Comment by mattmakesthings** [comment]
Hi Marko thanks for responding!
Im using a gtx 980 and have already made that change after looking over the past posts
![image](https://user-images.githubusercontent.com/25335149/29228532-edd16c06-7e8e-11e7-81eb-0671c606ce0d.png)

**Comment by mbjelonic** [comment]
Please undo your changes in the CMakeLists and try again. I already added the code generation for the compute capability 5.2. Can you please send me mor information if it still not working? E.g. Ubuntu version, ROS version and GNU version.

**Comment by mattmakesthings** [comment]
Still having same issue. I'm using: 
Ubuntu 16.04
ROS Kinetic
Linux version 4.10.0-30-generic
gcc version 5.4.0
Cuda compilation tools, release 8.0, V8.0.61

**Comment by mattmakesthings** [comment]
@mbjelonic would you mind posting what your versions of cuda, cmake, and gcc you were using?

**Comment by mbjelonic** [comment]
Hi, I tested darknet_ros on the following systems:

Ubuntu 14.04 and 16.04
ROS Kinetic and Indigo
Linux Version 4.10.0-27-generic
gcc version 5.4.0
Cuda compilation tools, release 8.0, V8.0.61
Graphics cards: Titan X (Compute capability 6.1) and Jetson Tx2 (Compute capability 6.2)
cmake version 3.5.1

Your system is similar to mine. You can try the following:

Build darknet_ros with CUDA and run it on your CPU. Simply copy and paste the following into your CMakeLists.txt:

`cmake_minimum_required(VERSION 2.8.12)
project(darknet_ros)

# Set c++11 cmake flags
set(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")

# Define path of darknet folder here.
find_path(DARKNET_PATH
  NAMES "README.md"
  HINTS "${CMAKE_CURRENT_SOURCE_DIR}/../darknet/")
message("Darknet path dir = ${DARKNET_PATH}")
add_definitions(-DDARKNET_FILE_PATH="${DARKNET_PATH}")
list(APPEND LIBRARIES "m")

# Find X11
MESSAGE ( STATUS "Searching for X11..." )
FIND_PACKAGE ( X11 REQUIRED )
IF ( X11_FOUND )
  INCLUDE_DIRECTORIES ( ${X11_INCLUDE_DIR} )
  LINK_LIBRARIES ( ${X11_LIBRARIES} )
  MESSAGE ( STATUS " X11_INCLUDE_DIR: " ${X11_INCLUDE_DIR} )
  MESSAGE ( STATUS " X11_LIBRARIES: " ${X11_LIBRARIES} )
ENDIF ( X11_FOUND )

# Find rquired packeges
find_package(Boost REQUIRED COMPONENTS thread)
find_package(OpenCV REQUIRED)
find_package(catkin REQUIRED
  COMPONENTS
    cv_bridge
    roscpp
    rospy
    std_msgs
    actionlib
    darknet_ros_msgs
    image_transport
)

# Enable OPENCV in darknet
add_definitions(-DOPENCV)
add_definitions(-O4 -g)

catkin_package(
  INCLUDE_DIRS
    include
  LIBRARIES
    ${PROJECT_NAME}_lib
  CATKIN_DEPENDS
    cv_bridge
    roscpp
    actionlib
    rospy
    std_msgs
    darknet_ros_msgs
    image_transport
  DEPENDS
    Boost
)

include_directories(
    ${DARKNET_PATH}/src
    ${DARKNET_PATH}/include
    include
    ${Boost_INCLUDE_DIRS}
    ${catkin_INCLUDE_DIRS}
)

add_library(${PROJECT_NAME}_lib
  src/YoloObjectDetector.cpp
  src/object_detector_demo.cpp
  src/image_interface.c

  ${DARKNET_PATH}/src/activation_layer.c        ${DARKNET_PATH}/src/im2col.c
  ${DARKNET_PATH}/src/activations.c             ${DARKNET_PATH}/src/image.c
  ${DARKNET_PATH}/src/avgpool_layer.c           ${DARKNET_PATH}/src/layer.c
  ${DARKNET_PATH}/src/batchnorm_layer.c         ${DARKNET_PATH}/src/list.c
  ${DARKNET_PATH}/src/blas.c                    ${DARKNET_PATH}/src/local_layer.c
  ${DARKNET_PATH}/src/box.c                     ${DARKNET_PATH}/src/lstm_layer.c
  ${DARKNET_PATH}/src/col2im.c                  ${DARKNET_PATH}/src/matrix.c
  ${DARKNET_PATH}/src/compare.c                 ${DARKNET_PATH}/src/maxpool_layer.c
  ${DARKNET_PATH}/src/connected_layer.c         ${DARKNET_PATH}/src/network.c
  ${DARKNET_PATH}/src/convolutional_layer.c     ${DARKNET_PATH}/src/normalization_layer.c
  ${DARKNET_PATH}/src/cost_layer.c              ${DARKNET_PATH}/src/option_list.c
  ${DARKNET_PATH}/src/crnn_layer.c              ${DARKNET_PATH}/src/parser.c
  ${DARKNET_PATH}/src/crop_layer.c              ${DARKNET_PATH}/src/region_layer.c
  ${DARKNET_PATH}/src/cuda.c                    ${DARKNET_PATH}/src/reorg_layer.c
  ${DARKNET_PATH}/src/data.c                    ${DARKNET_PATH}/src/rnn_layer.c
  ${DARKNET_PATH}/src/deconvolutional_layer.c   ${DARKNET_PATH}/src/route_layer.c
  ${DARKNET_PATH}/src/demo.c                    ${DARKNET_PATH}/src/shortcut_layer.c
  ${DARKNET_PATH}/src/detection_layer.c         ${DARKNET_PATH}/src/softmax_layer.c
  ${DARKNET_PATH}/src/dropout_layer.c           ${DARKNET_PATH}/src/tree.c
  ${DARKNET_PATH}/src/gemm.c                    ${DARKNET_PATH}/src/utils.c
  ${DARKNET_PATH}/src/gru_layer.c

  ${DARKNET_PATH}/examples/art.c                ${DARKNET_PATH}/examples/regressor.c
  ${DARKNET_PATH}/examples/captcha.c            ${DARKNET_PATH}/examples/rnn.c
  ${DARKNET_PATH}/examples/cifar.c              ${DARKNET_PATH}/examples/rnn_vid.c
  ${DARKNET_PATH}/examples/classifier.c         ${DARKNET_PATH}/examples/segmenter.c
  ${DARKNET_PATH}/examples/coco.c               ${DARKNET_PATH}/examples/super.c
  ${DARKNET_PATH}/examples/detector.c           ${DARKNET_PATH}/examples/swag.c
  ${DARKNET_PATH}/examples/dice.c               ${DARKNET_PATH}/examples/tag.c
  ${DARKNET_PATH}/examples/go.c                 ${DARKNET_PATH}/examples/voxel.c
  ${DARKNET_PATH}/examples/lsd.c                ${DARKNET_PATH}/examples/writing.c
  ${DARKNET_PATH}/examples/nightmare.c          ${DARKNET_PATH}/examples/yolo.c
)

target_link_libraries(${PROJECT_NAME}_lib
  m
  pthread
  stdc++
  ${Boost_LIBRARIES}
  ${catkin_LIBRARIES}
)

add_executable(${PROJECT_NAME}
  src/yolo_object_detector_node.cpp
)

target_link_libraries(${PROJECT_NAME}
  ${PROJECT_NAME}_lib
)

install(
  TARGETS ${PROJECT_NAME}_lib
  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
)

install(
  DIRECTORY include/${PROJECT_NAME}/
  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
  FILES_MATCHING PATTERN "*.h"
)

# Download tiny-yolo-voc.weights
set(PATH "${CMAKE_CURRENT_SOURCE_DIR}/yolo_network_config/weights")
set(FILE "${PATH}/tiny-yolo-voc.weights")
message(STATUS "Checking and downloading tiny-yolo-voc.weights if needed ...")
if (NOT EXISTS "${FILE}")
  message(STATUS "... file does not exist. Downloading now ...")
  execute_process(COMMAND wget http://pjreddie.com/media/files/tiny-yolo-voc.weights -P ${PATH})
endif()

#############
## Testing ##
#############

if(CATKIN_ENABLE_TESTING)
  # Download yolo-voc.weights
  set(PATH "${CMAKE_CURRENT_SOURCE_DIR}/yolo_network_config/weights")
  set(FILE "${PATH}/yolo-voc.weights")
  message(STATUS "Checking and downloading yolo-voc.weights if needed ...")
  if (NOT EXISTS "${FILE}")
    message(STATUS "... file does not exist. Downloading now ...")
    execute_process(COMMAND wget http://pjreddie.com/media/files/yolo-voc.weights -P ${PATH})
  endif()

  find_package(rostest REQUIRED)

  # Object detection in images.
  add_rostest_gtest(${PROJECT_NAME}_object_detection-test
    test/object_detection.test
    test/test_main.cpp
    test/ObjectDetection.cpp
  )
  target_link_libraries(${PROJECT_NAME}_object_detection-test
    ${catkin_LIBRARIES}
  )
endif()`

**Comment by mattmakesthings** [comment]
So it ended up actually being an error with my cuda installation

I followed these steps and got darknet_ros to run
https://covijn.com/2016/11/fixing-darknet-opencv3-make-error-convolutional_kernels/

**Comment by mbjelonic** [comment]
Thanks for the update!

**Comment by He-Rong** [comment]
I met the same problem and I am wondering more details to solve the problem. Did you reinstall cuda?

**Comment by cardboardcode** [comment]
@mbjelonic thanks, your suggestion worked for me. 

```
Build darknet_ros without CUDA and run it on your CPU. 
Simply remove the following line in the CMakeLists.txt:

FIND_PACKAGE(CUDA)
```
To be more specific, please comment out the lines **16** and **17** of  `CMakeLists.txt` under `darknet_ros` ROS package.

line 16: [find package(CUDA QUIET)](https://github.com/leggedrobotics/darknet_ros/blob/ac666ab8e8e3dd23a8a95d891fb90874e63c8cb5/darknet_ros/CMakeLists.txt#L16)
line 18: [find package(CUDA REQUIRED)](https://github.com/leggedrobotics/darknet_ros/blob/ac666ab8e8e3dd23a8a95d891fb90874e63c8cb5/darknet_ros/CMakeLists.txt#L18)
